### ISSUE TITLE
Dataset deprecation guidelines result in errors

### ISSUE THREAD

**Issue by andregrageda** [Bioconductor/pkgrevdocs#120]
Hi! I've tried to implement dataset deprecation utilizing the guidelines found [here](https://contributions.bioconductor.org/deprecation.html#deprecate-dataset).

Running this code, results in the following error:

```
pkgDataset <- data.frame(1:10)
usethis::use_data(pkgDataset)

delayedAssign(
    x = "pkgDataset",
    value = {
      warning("'pkgDataset' dataset is deprecated; see '?newData'")
      pkgDataset
    }
  )

pkgDataset
```

```
Error: promise already under evaluation: recursive default argument reference or earlier problems?
In addition: There were 19 warnings (use warnings() to see them)
```

My workaround was to create another dataset `pkgDataset_deprecated` and call that in `delayedAssign` instead.

```
pkgDataset <- data.frame(1:10)
pkgDataset_deprecated <- pkgDataset
usethis::use_data(pkgDataset_deprecated)

delayedAssign(
  x = "pkgDataset",
  value = {
    warning("'pkgDataset' dataset is deprecated; see '?newData'")
    pkgDataset_deprecated
  }
)

pkgDataset
```

I'd be happy to submit a PR but it looks like I don't have permission. Let me know if you think this solution makes sense, thanks!

**Comment by LiNk-NY** [comment]
Hi Andre, @andregrageda 

Sorry I just saw this!
The order of operations does not seem correct. 
Previously, it should have been : 

```r
pkgDataset <- data.frame(1:10)
delayedAssign(
    x = "pkgDataset",
    value = {
      warning("'pkgDataset' dataset is deprecated; see '?newData'")
      pkgDataset
    }
  )
usethis::use_data(pkgDataset)
```

But now, for 4.4.0 and 4.3.x patched, the new recommendations are in  
PR https://github.com/Bioconductor/pkgrevdocs/pull/121.

**Comment by andregrageda** [comment]
> Hi Andre, @andregrageda
> 
> Sorry I just saw this! The order of operations does not seem correct. Previously, it should have been :
> 
> ```r
> pkgDataset <- data.frame(1:10)
> delayedAssign(
>     x = "pkgDataset",
>     value = {
>       warning("'pkgDataset' dataset is deprecated; see '?newData'")
>       pkgDataset
>     }
>   )
> usethis::use_data(pkgDataset)
> ```
> 
> But now, for 4.4.0 and 4.3.x patched, the new recommendations are in PR #121.

No worries, thanks!
